BDArmory/Control/MissileFire.cs

```diff
@@ -1076,15 +1076,18 @@ public IBDWeapon selectedWeapon
         {
             get
             {
-                if (sw != null)
+                if (sw != null) //we have a weapon set by bool SmartPick, or set by the last time this was called
                 {
                     var gun = sw.GetWeaponModule();
-                    if (gun == null || (gun.useThisWeaponForAim && (gun.ammoCount > 0 || BDArmorySettings.INFINITE_AMMO)))
+                    if (gun == null || (gun.useThisWeaponForAim && (gun.ammoCount > 0 || BDArmorySettings.INFINITE_AMMO))) //it's a missile, or an aim-override enabled weapon with ammo
                         if (sw.GetPart().vessel == vessel) return sw;
+                        else sw = null; //weapon no longer on craft. Null in case below while loop doesn't find other weapons in the same group on craft.
```

Comment:
I think it's better to just use a blanket
```csharp
sw = null; // Null in case below while loop doesn't find other weapons in the same group on craft.
```
at the same point that
```csharp
IBDWeapon candidateGun = null;
```
is called, as that avoids any potential loop-holes in the logic for `null`ing it here, e.g., if the currently selected gun has fallen off the vessel and either doesn't have the aiming override or is out of ammo, then it won't be `null`ed here.